src/main/scala/cromwell/engine/StoreActor.scala

```diff
@@ -0,0 +1,100 @@
+  case class UpdateStatus(call: Call, status: ExecutionStatus.Value) extends StoreActorMessage
+  case object GetOutputs extends StoreActorMessage
+  case class GetLocallyQualifiedInputs(call: Call) extends StoreActorMessage
+  implicit val ActorTimeout = Timeout(5 seconds)
+}
+
+/** Actor to hold symbol and execution status data for a single workflow. */
+class StoreActor(binding: WdlBinding, inputs: WorkflowCoercedInputs) extends Actor {
+  private val symbolStore = new SymbolStore(binding, inputs)
+  private val executionStore = new ExecutionStore(binding)
+  private val log = Logging(context.system, this)
+
+  override def receive: Receive = LoggingReceive {
+
+    case CallCompleted(call, callOutputs) =>
+      handleCallCompleted(call, callOutputs) pipeTo sender
+
+    case FindRunnableCalls =>
+      sender ! executionStore.runnableCalls
+
+    case PrepareToStartCalls(calls) =>
+      handlePrepareToStartCalls(calls) pipeTo sender
+
+    case GetOutputs =>
+      sender ! (symbolStore.getOutputs map symbolStoreEntryToMapEntry).toMap
+
+    case GetLocallyQualifiedInputs(call) =>
+      sender ! symbolStore.locallyQualifiedInputs(call)
+
+    case UpdateStatus(call, status) =>
+      executionStore.updateStatus(call, status)
+  }
+
+  private def symbolStoreEntryToMapEntry(e: SymbolStoreEntry): (String, WdlValue) =
+    e.key.scope + "." + e.key.name -> e.wdlValue.get
+
+  private def updateOutputs(call: Call, callOutputs: Map[String, WdlValue]): Future[Unit] = Future {
+
+    def addOutputValueToSymbolStore(callOutput: (String, WdlValue)): Try[Unit] =
+      symbolStore.addOutputValue(call.fullyQualifiedName, callOutput._1, Some(callOutput._2), callOutput._2.wdlType)
+
+    val addedEntries = callOutputs map addOutputValueToSymbolStore
+    val failureMessages = TryUtil.stringifyFailures(addedEntries)
+
+    if (failureMessages.isEmpty) {
+      executionStore.updateStatus(call, ExecutionStatus.Done)
+    } else {
+      val errorMessages = failureMessages.mkString("\n")
+      log.error(errorMessages)
+      executionStore.updateStatus(call, ExecutionStatus.Failed)
+      throw new RuntimeException(errorMessages)
+    }
+  }
+
+  private def handleCallCompleted(call: Call, callOutputs: Map[String, WdlValue]): Future[Boolean] =
+    for {
+      _ <- updateOutputs(call, callOutputs)
+    } yield executionStore.isDone
+
+  private def handlePrepareToStartCalls(calls: Iterable[Call]): Future[Unit] = Future {
+    calls.foreach { call =>
+      call.inputMappings.foreach { case (inputName, expression) =>
+        val ast = expression.ast.asInstanceOf[Ast]
+        val Seq(lhs, rhs) = Seq("lhs", "rhs").map {
+          ast.getAttribute(_).asInstanceOf[Terminal].getSourceString
+        }
+        val outputFqn = Seq(call.parent.get.name, lhs, rhs).mkString(".")
+        val inputFqn = Seq(call.parent.get.name, call.name, inputName).mkString(".")
+        symbolStore.copyOutputToInput(outputFqn, inputFqn)
```

Comment:
Nothing to do with this PR but this did remind me why we filed that tech debt ticket about call input expressions :)